internal/cmd/test.go

```diff
@@ -135,3 +137,43 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+type CommonCMDTestCases struct {
+	ExpectedExitCode int
+	ExpectedStdout   string
+	ExpectedStderr   string
+}
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, commonCases CommonCMDTestCases, prefix string) (string, string, int) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		test.Errorf("%sCMD run returned an error: '%v'.", prefix, err)
```

Comment:
We talked about this a previous round.

We don't want our test failures to cascade.
You need to return on every `test.Errorf()`.